src/Main.cpp

```diff
@@ -75,26 +101,60 @@ class HostUartCommTask : public AbstractTask<2 * configMINIMAL_STACK_SIZE> {
   private:
     void task() override {
         IHostUart& hostUart = BSPContainer::getHostUart();
+        uint8_t buffer[5];
+        buffer[sizeof(buffer) - 1] = 0;
+
         while (true) {
             hostUart.send((const uint8_t*)"HELLO WORLD", sizeof("HELLO WORLD"));
             vTaskDelay(1000);
         }
     }
 };
 
+class TCPReadDemoTask : public AbstractTask<2 * configMINIMAL_STACK_SIZE> {
+  public:
+    TCPReadDemoTask(const char* taskName, UBaseType_t priority) :
+        AbstractTask(taskName, priority), m_logger(LoggerContainer::getLogger()) {}
+
+    ~TCPReadDemoTask() override = default;
+
+  private:
+    ILogger& m_logger;
+    void task() override {
+
+        uint8_t buffer[5];
+        buffer[sizeof(buffer) - 1] = 0;
+
+        // Wait for connection
+        while (true) {
+            if (std::optional<TCPClientWrapper> socket = SocketContainer::getHostClientSocket()) {
+                while (true) {
+                    auto ret = socket.value().receive(buffer, sizeof(buffer) - 1);
+                    (void)ret;
```

Comment:
what is this line doing? (for my understanding)

Reply:
@t-constantin  It's simly to remove the compiler warning about unused variable,  we have a LOT of compiler checks on the hivemind.
Though removing the whole ret variable would fix this problem (thus ignoring the return) and remove the need of the explicit discard
`socket.value().receive(buffer, sizeof(buffer) - 1);`

Reply:
it was to prevent an error being thrown because of an unused parameter (because I needed to debug the value) but it is not needed anymore

Reply:
I see, thanks